libcxx/test/std/ranges/range.adaptors/range.join/range.join.iterator/eq.pass.cpp

```diff
@@ -43,16 +42,14 @@ constexpr bool test() {
   }
 
   {
-    // !equality_comparable<iterator_t<Base>>
+    // !forward_range<iterator_t<Base>>
     using Inner = BufferView<int*>;
     using Outer = BufferView<cpp20_input_iterator<Inner*>, sentinel_wrapper<cpp20_input_iterator<Inner*>>>;
     static_assert(!std::equality_comparable<std::ranges::iterator_t<Outer>>);
     Inner inners[2] = {buffer[0], buffer[1]};
     std::ranges::join_view jv(Outer{inners});
     auto iter = jv.begin();
     static_assert(!std::equality_comparable<decltype(iter)>);
-    auto const_iter = std::as_const(jv).begin();
```

Comment:
I don't understand why this isn't valid anymore?

Reply:
This is not valid anymore, because `jv` adapts input range (of type `Outer`) and, since P2770R0, `begin() const` requires forward range.

Reply:
I've noticed that the comment above is wrong, it should say `!forward_range<Base>` instead of `!forward_range<iterator_t<Base>>`.